apis/python/src/tiledb/vector_search/vamana_index.py

```diff
@@ -0,0 +1,154 @@
+import json
+import multiprocessing
+from typing import Any, Mapping
+
+import numpy as np
+from tiledb.cloud.dag import Mode
+
+from tiledb.vector_search import index
+from tiledb.vector_search.module import *
+from tiledb.vector_search.storage_formats import (STORAGE_VERSION,
+                                                  storage_formats,
+                                                  validate_storage_version)
+from tiledb.vector_search.utils import add_to_group
+from tiledb.vector_search import _tiledbvspy as vspy
+
+MAX_UINT64 = np.iinfo(np.dtype("uint64")).max
+INDEX_TYPE = "VAMANA"
+
+class VamanaIndex(index.Index):
+    """
+    Open a Vamana index
+
+    Parameters
+    ----------
+    uri: str
+        URI of the index
+    config: Optional[Mapping[str, Any]]
+        config dictionary, defaults to None
+    memory_budget: int
+        Main memory budget. If not provided, no memory budget is applied.
+    """
+
+    def __init__(
+        self,
+        uri: str,
+        config: Optional[Mapping[str, Any]] = None,
+        timestamp=None,
+        **kwargs,
+    ):
+        super().__init__(uri=uri, config=config, timestamp=timestamp)
+        self.index_type = INDEX_TYPE
+        self.index = vspy.IndexVamana(vspy.Ctx(config), uri)
+        self.db_uri = self.group[
+            storage_formats[self.storage_version]["PARTS_ARRAY_NAME"]
+            + self.index_version
```

Comment:
I don't think that `index_version` is needed for this new type of index. It is used only for backwards compatibility for the existing indexes.

Reply:
Thanks, done.